File: frontend/types/expense.ts
export interface Expense {
  id: number;
  amount: number;
  description: string;
  category: string;
  date: string;
  created_at: string;
  updated_at: string;
}

export interface ExpenseSummary {
  total: number;
  count: number;
}

export const CATEGORIES = [
  'makanan',
  'transport',
  'belanja',
  'hiburan',
  'lainnya'
] as const;

export type Category = typeof CATEGORIES[number];

File: backend/models/expense.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Expense struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	Amount      float64   `json:"amount" gorm:"not null"`
	Description string    `json:"description" gorm:"type:text"`
	Category    string    `json:"category" gorm:"type:text;not null"`
	Date        time.Time `json:"date" gorm:"not null"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}

